src/lib.rs

```diff
@@ -418,6 +369,74 @@ impl Game {
         Ok(())
     }
 
+    fn kill_player_near(&mut self, position: Position) -> Result<(), JsValue> {
+        let local_player_color = match &self.local_player_color {
+            None => return Ok(()), // not controlling anything
+            Some(c) => *c,
+        };
+
+        let mut killed_player: Option<&mut Player> = None;
+        let mut closest_distance = self.kill_distance;
+
+        for player in self.players.iter_mut() {
+            // TODO(can't kill impostors)
+            if player.color == local_player_color || player.dead {
+                continue;
+            }
+
+            let distance = position.distance(player.position);
+            if distance < closest_distance {
+                killed_player = Some(player);
+                closest_distance = distance;
+            }
+        }
+
+        if let Some(player) = killed_player {
+            player.dead = true;
+            let player: &Player = player;
+            let body = DeadBody {
+                position: player.position,
+                color: player.color,
+            };
+            self.bodies.push(body);
+            self.send_msg(&Message::Killed(body))?;
+            // Move the killer on top of the new body.
+            if let Some(player) = self.local_player_mut() {
+                player.position = body.position;
+            }
+        }
+
+        Ok(())
+    }
+
+    fn activate_near(&mut self, position: Position) -> Result<(), JsValue> {
+        let mut closest_distance = self.kill_distance;
+        let local_player = match self.local_player_mut() {
+            Some(player) => player,
+            None => return Ok(()),
+        };
+        let is_imp = local_player.impostor;
+        let color = local_player.color;
+
+        let mut finished_task: Option<(usize, &mut Task)> = None;
+        for (i, task) in local_player.tasks.iter_mut().enumerate() {
+            // TODO(can't kill impostors)
+
+            let distance = position.distance(task.position);
+            if distance < closest_distance {
+                finished_task = Some((i, task));
+                closest_distance = distance;
+            }
+        }
+        if let Some((i, task)) = finished_task {
```

Comment:
It took my brain a second to accept that this was intentional assignment. Something about the `Some` being on the LHS disrupted my expectations...

Reply:
Yeah, `if let ...` pattern matching takes some getting used to. I'm currently in favor of it, but I'm definitely doing more writing than reading at the moment